src/dash-playlist-loader.js

```diff
@@ -220,7 +226,20 @@ export default class DashPlaylistLoader extends EventTarget {
     setupMediaPlaylists(master);
     resolveMediaGroupUris(master);
 
-    return master;
+    let hasSidx = false;
+    for (var key in master.playlists) {
+      if (master.playlists[key].sidx) {
+        hasSidx = true;
+        break;
+      }
+    }
+
+    if (hasSidx) {
+      // Note: this calls done!
+      this.fetchMediaSegmentsFromSidx_(master.playlists, master, done);
```

Comment:
There should be a return here otherwise `done` will get called twice when `hasSidx` is true